File: VDFEditor/src/test/java/de/treichels/hott/vdfeditor/actions/AddActionTest.java
package de.treichels.hott.vdfeditor.actions;

import static org.junit.Assert.assertEquals;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import org.junit.Before;
import org.junit.Test;

import de.treichels.hott.vdfeditor.actions.AddAction;
import de.treichels.hott.vdfeditor.actions.UndoableAction;

public class AddActionTest {
    private List<String> list;
    private UndoableAction<String> action;

    @Before
    public void setup() {
        list = new ArrayList<>(Arrays.asList("a", "b", "c", "d"));
    }

    @Test
    public void testAddAction() {
        action = new AddAction<>("x");
        assertEquals("[a, b, c, d]", list.toString());
        action.apply(list);
        assertEquals("[a, b, c, d, x]", list.toString());
        action.undo(list);
        assertEquals("[a, b, c, d]", list.toString());

    }
}

File: VDFEditor/src/test/kotlin/de/treichels/hott/vdfeditor/actions/AddActionTest.kt
package de.treichels.hott.vdfeditor.actions

import org.junit.Test
import kotlin.test.assertEquals

class AddActionTest : AbstractActionTest() {
    @Test
    fun testAddAction() {
        val action = AddAction("x")
        assertEquals("[a, b, c, d]", list.toString())
        action.apply(list)
        assertEquals("[a, b, c, d, x]", list.toString())
        action.undo(list)
        assertEquals("[a, b, c, d]", list.toString())
    }
}
